packages/@aws-cdk/aws-rds/lib/cluster-engine.ts

```diff
@@ -389,18 +423,22 @@ export class AuroraPostgresEngineVersion {
    * @param auroraPostgresMajorVersion the major version of the engine,
    *   for example "9.6"
    */
-  public static of(auroraPostgresFullVersion: string, auroraPostgresMajorVersion: string): AuroraPostgresEngineVersion {
-    return new AuroraPostgresEngineVersion(auroraPostgresFullVersion, auroraPostgresMajorVersion);
+  public static of(auroraPostgresFullVersion: string, auroraPostgresMajorVersion: string,
```

Comment:
I have a slightly different idea here.

Since the feature name is always `s3Import` / `s3Export`, how about we do this?

```ts
export interface AuroraPostgresEngineFeatures {
  readonly s3ImportFeature?: boolean;
  readonly s3ExportFeature?: boolean;
}

export class AuroraPostgresEngineVersion {
  public static of(auroraPostgresFullVersion: string, auroraPostgresMajorVersion: string, features?: AuroraPostgresEngineFeatures) {
    // ..
  }
}
```

And then default the feature names `s3Import` / `s3Export` inside the implementation of `AuroraPostgresEngineVersion` - much less duplication.

If, in the future, there is a new feature name for S3 Import / S3 Export, we can always add more properties to `AuroraPostgresEngineFeatures`.

Reply:
yeah I was actually doing exactly this after I'd put it into draft :) - glad to have some validation that it's the way to DRY this up